src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -65,31 +71,67 @@ public IReadOnlyList<ILibraryGroup> AvailablePackages
 
         public IReadOnlyList<PackageItem> DisplayRoots
         {
-            get { return _displayRoots; }
-            set { Set(ref _displayRoots, value); }
+            get
+            {
+                IReadOnlyList<PackageItem> displayRoots = _displayRoots;
+
+                if (displayRoots != null && displayRoots.Any())
+                {
+                    displayRoots.ElementAt(0).Name = Text.Files;
+                }
+
+                return _displayRoots;
+            }
+
+            set
+            {
+                if (String.IsNullOrEmpty(PackageId))
+                {
+                    Set(ref _displayRoots, null);
+                }
+                else
+                {
+                    Set(ref _displayRoots, value);
+                }
+            }
         }
 
         public ICommand InstallPackageCommand { get; }
 
-        public bool IsTreeViewEmpty => SelectedPackage == null;
+        public bool IsTreeViewEmpty
+        {
+            get { return _isTreeViewEmpty; }
+            set
+            {
+                if (Set(ref _isTreeViewEmpty, value))
+                {
+                    RefreshFileSelections();
+                }
+            }
+        }
 
         public string PackageId
         {
             get { return _packageId; }
             set
             {
-                if (Set(ref _packageId, value))
+                // If libraryId is null, then we need to clear the tree view for files and show warning message.
+                if (String.IsNullOrEmpty(value))
                 {
-                    SelectedProvider.GetCatalog().GetLibraryAsync(_packageId, CancellationToken.None).ContinueWith(x =>
+                    if (Set(ref _packageId, value))
                     {
-                        if (x.IsFaulted || x.IsCanceled)
+                        SelectedProvider.GetCatalog().GetLibraryAsync(_packageId, CancellationToken.None).ContinueWith(x =>
```

Comment:
What does GetLibraryAsync do when _packageId is null or empty?

Reply:
For cdnjs, Asset would be null and we throw InvalidLibraryException: https://github.com/aspnet/LibraryManager/blob/baf5b4bc5cdcdf21feda349c248c7e5c0a81ee7f/src/LibraryManager/Providers/Cdnjs/CdnjsCatalog.cs#L139
For fileSystem, we don't do any validation and return empty library: https://github.com/aspnet/LibraryManager/blob/2bd94030d2d082efc8167a6b719132447e86c9ae/src/LibraryManager/Providers/FileSystem/FileSystemCatalog.cs#L136

I removed the call to GetLibraryAsync since it isn't useful from UI perspective. When _packageId is empty, I clear the state for treeview(selected files) and display warning message. So this should be sufficient: https://github.com/aspnet/LibraryManager/commit/43ab2c219d509a016c4abe839aec509ab9725def

Let me know if you have any suggestions.